lib/3scale/backend/transactor/report_job.rb

```diff
@@ -8,8 +8,11 @@ class ReportJob < BackgroundJob
 
         class << self
           def perform_logged(service_id, raw_transactions, _enqueue_time, context_info = {})
-            transactions = parse_transactions(service_id, raw_transactions)
-            ProcessJob.perform(transactions) if !transactions.nil? && transactions.size > 0
+            context_info ||= {} # avoid nils potentially existing in older versions
+            request_info = context_info['request'.freeze] || {}
+
+            transactions = parse_transactions(service_id, raw_transactions, request_info)
+            ProcessJob.perform(transactions, context_info) if !transactions.nil? && transactions.size > 0
```

Comment:
The context info sent here is not used. Actually, `ProcessJob.perform` receives a hash of options but it only cares about whether it's a report for the master account. Maybe we should change that to simply receive a boolean.

Reply:
This was done this way because it follows a similar structure to other jobs' `.perform` methods, and avoids tying us to the specific usage the job makes of the context info. It's not a big deal, the work has to be done either there or here, but doing it here means knowing about what `ProcessJob` does inside.